### PR TITLE
Update hack/update-deps.sh to drop pruned transport/grpc

### PR DESCRIPTION
**Pull request by fejta**
/assign @stevekuznetsov @BenTheElder 

The basic pattern when you see this error message:
```
ERROR: /prefix/src/k8s.io/test-infra/vendor/github.com/golang/protobuf/protoc-gen-go/BUILD:3:1: no such package 'vendor/github.com/golang/protobuf/protoc-gen-go/generator': BUILD file not found on package path and referenced by '//vendor/github.com/golang/protobuf/protoc-gen-go:go_default_library'
```

Is to notice these two lines:
`no such package 'vendor/github.com/golang/protobuf/protoc-gen-go/generator': BUILD file not found`
and
` referenced by '//vendor/github.com/golang/protobuf/protoc-gen-go:go_default_library'`

And convert this into a `drop-dep vendor/github.com/golang/protobuf/protoc-gen-go/generator vendor/github.com/golang/protobuf/protoc-gen-go` line in update-deps.sh

I should be able to automate this, but in the mean time that's the process

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
```
+ patch-text-internal
hack/update-deps.sh: line 58: patch-text-internal: command not found
```

File: hack/update-deps.sh [hunk 1]
```diff
@@ -36,20 +37,24 @@ set -o xtrace
...
 main() {
   pushd "$(dirname "${BASH_SOURCE}")/.."
   dep ensure
   dep prune
   hack/update-bazel.sh
+  drop-dep vendor/golang.org/x/text/language vendor/golang.org/x/text/internal
+  drop-dep vendor/google.golang.org/api/transport/grpc vendor/google.golang.org/api/transport
+  drop-dep vendor/github.com/golang/protobuf/protoc-gen-go/grpc vendor/github.com/golang/protobuf/protoc-gen-go
+  drop-dep vendor/github.com/golang/protobuf/protoc-gen-go/generator vendor/github.com/golang/protobuf/protoc-gen-go
   patch-text-internal  # TODO(fejta): fix dep prune instead
```
Comment (stevekuznetsov) [inline hack/update-deps.sh]: Get rid of this line?

Comment (fejta) [reply hack/update-deps.sh]: Done

**Comment by BenTheElder** [comment]
More bash oh-my! :scream:

**Pull request merged**